hyperspy/_signals/eds_tem.py

```diff
@@ -267,6 +267,10 @@ def get_calibration_from(self, ref, nb_pix=1):
         # Setup the axes_manager
         ax_m = self.axes_manager.signal_axes[0]
         ax_ref = ref.axes_manager.signal_axes[0]
+        for _axis in [ax_m, ax_ref]:
+            if not _axis.is_uniform:
+                raise NotImplementedError(
+                    "The function is not implemented for non-uniform axes.")
```

Comment:
Would apply to quite a number of functions. I guess if it is not on the bucket list of things to solve soon (like split and rebin) it would make sense. On the other hand, there is always the danger that once we fix it, it is forgotten in the docstring.

Reply:
Yes, it is not strictly necessary but I think that it would be good to add it here to have the docstring as good as possible. We strongly encourage users to read docstring, so they need to be of the highest standard as possible! ;)

Reply:
I added it to all relevant docstrings, where a NotImplementedError for non-uniform axes is raised.